stockApp/stockApp/View/AssetViewController.swift

```diff
@@ -0,0 +1,84 @@
+import UIKit
+
+//MARK: - Class
+
+class AssetViewController: UIViewController {
+
+    //MARK: - Properties
+    
+    lazy var superView: AssetView = {
+        let view = AssetView(frame: UIScreen.main.bounds)
+        view.delegate = self
+        view.assetsListUITableView.delegate = self
+        view.assetsListUITableView.dataSource = self
+        view.assetsListUITableView.register(AssetTableViewCell.self, forCellReuseIdentifier: Strings.cellIdentifier)
+        return view
+    }()
+    
+    lazy var viewModel: AssetViewModelProtocol = {
+        return AssetViewModel(useCase: GetAssetUseCase(manager: AssetManager()))
+    }()
+    
+    var coordinator: CoordinatorProtocol?
+    
+    var assets: [Asset] = []
+    
+    //MARK: - Setup
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setUpViewModel()
+        setUpUI()
+        self.navigationController?.isNavigationBarHidden = false
+        self.navigationController?.navigationBar.tintColor = Color.porcelainWhite
+        self.navigationController?.navigationBar.barTintColor = Color.atlantisGreen
+        viewModel.retrieveData(filter: "", value: "", onCompletion: {})
+    }
+    
+    func setUpViewModel() {
+        viewModel.observerData = { assets in
+            self.assets = assets
+            self.displayData()
+         }
+    }
+    
+    func setUpUI() {
+        view.addSubview(superView)
+    }
+    
+    func displayData() {
+        DispatchQueue.main.async {
+            guard let searchView = self.view as? AssetView else {
+                return
+            }
+            searchView.assetsListUITableView.reloadData()
+        }
+    }
+}
+
+extension AssetViewController: AssetViewDelegate {
+    
+    func searchButtonPressed(from view: AssetView) {
+        HelperFunctions.showSearchAlertMessage(title: Strings.searchTitleMessage,viewController: self)
+    }
+}
+
+extension AssetViewController: UITableViewDelegate, UITableViewDataSource {
+    
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return assets.count
+    }
+    
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        guard let cell = tableView.dequeueReusableCell(withIdentifier: Strings.cellIdentifier, for: indexPath) as? AssetTableViewCell else {
+            return AssetTableViewCell()
+        }
+        cell.setFields(barcode: assets[indexPath.row].barcode, assetName: assets[indexPath.row].assetName, globerMail: assets[indexPath.row].globerMail, site: assets[indexPath.row].site)
```

Comment:
Dont repeat yourself


```suggestion
        let model = assets[indexPath.row]
        cell.setFields(barcode: model.barcode, 
                               assetName: model.assetName, 
                               globerMail: model.globerMail, 
                               site: model.site)
```